src/pages/Florist/index.tsx

```diff
@@ -200,69 +205,128 @@ export const Florist = () => {
+                      className={classes.boughtName}
                     >
-                      ???
+                      {flower.name}
                     </Typography>
                     <img
                       src={flower.imageURL}
-                      alt={"secret flower pic"}
-                      className={styles.notBoughtPic}
+                      alt={`${flower.name} pic`}
+                      className={styles.boughtPic}
                     />
-                    <Button
-                      variant="contained"
-                      onClick={() => buyFlowerHandler(flower._id, flower.price)}
-                      color="primary"
-                      className={classes.buyButton}
-                      disabled={
-                        !userData.balance || flower.price > userData.balance
-                          ? true
-                          : false
+                  </Grid>
+                ) : (
+                  <Grid
+                    container
+                    direction="column"
+                    alignItems="center"
+                    justifyContent="center"
+                    className={classes.card}
+                    key={flower._id}
+                  >
+                    <LoadingWrapper
+                      isLoading={
+                        flower._id === lastBought &&
+                        buyFlowerAPIState.status === "loading"
                       }
                     >
-                      Buy: {flower.price}
-                    </Button>
-                  </LoadingWrapper>
-                </Grid>
-              );
-            })}
+                      <Typography
+                        variant="caption"
+                        className={classes.notBoughtName}
+                      >
+                        ???
+                      </Typography>
+                      <img
+                        src={flower.imageURL}
+                        alt={"secret flower pic"}
+                        className={styles.notBoughtPic}
+                      />
+                      <Button
+                        variant="contained"
+                        onClick={() => {
+                          setFloristStep(1);
+                          setSelectFlowerData(flower);
+                          // buyFlowerHandler(flower._id, flower.price);
+                        }}
+                        color="primary"
+                        className={classes.buyButton}
+                        disabled={
+                          !userData.balance || flower.price > userData.balance
+                            ? true
+                            : false
+                        }
+                      >
+                        Buy: {flower.price}
+                      </Button>
+                    </LoadingWrapper>
+                  </Grid>
+                );
+              })}
+            {floristStep === 1 && (
```

Comment:
It would be better to use a modal for this,  also please separate this into a different component
I'd like to discuss tomorrow about the implementation

Reply:
memo  : modal
https://material-ui.com/components/modal/

Reply:
https://material-ui.com/components/dialogs/